dss-api.yml

```diff
@@ -702,6 +702,118 @@ paths:
                     enum: [unhandled_exception, illegal_arguments, bundle_already_exists]
                 required:
                   - code
+  /bundles/{uuid}/checkout:
+    post:
+      summary: Check out a bundle to HCA-owned cloud storage or user provided destination
+      description: |
+        User makes a request for a bundle version to be extracted. Storage system makes an attempt to validate
+        that the bundle UUID with an optionally provided version exists in a given replica.
+        If the storage service fails to locate the bundle, the error is immediately raised to the user.
+        Otherwise start copying the data in the background to the hot storage.
+        TODO: After some time period, the data will be removed.
+        TBD: This could be based on initial checkout time or last access time.
+      operationId: dss.api.bundles.checkout.post
+      parameters:
+        - name: uuid
+          in: path
+          description: A RFC4122-compliant ID for the bundle.
+          required: true
+          type: string
+          pattern: "[A-Za-z0-9]{8}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{4}-[A-Za-z0-9]{12}"
+        - name: version
+          in: query
+          description: Timestamp of file creation in RFC3339.  If this is not provided, the latest version is returned.
+          required: false
+          type: string
+          format: date-time
+        - name: replica
+          in: query
+          description: Replica to fetch from.
+          required: true
+          type: string
+          enum: [aws, gcp, azure]
+        - name: json_request_body
+          in: body
+          required: true
+          schema:
+            type: object
+            properties:
+              email:
+                description: An email address to send status updates to.
+                type: string
+                format: email
+              destination:
+                description: User-owned destination storage bucket.
+                type: string
+            required:
+              - email
+      responses:
+        200:
+          description: Returned when the bundle UUID with optionally specified version exists and checkout has been initiated.
+          schema:
+            type: object
+            properties:
+              checkout_job_id:
+                description: Unique identifier of the checkout job request.
+                type: string
+            required:
+              - checkout_job_id
+        default:
+          description: Unexpected error
+          schema:
+            allOf:
+              - $ref: '#/definitions/Error'
+              - type: object
+                properties:
+                  code:
+                    type: string
+                    description: Machine-readable error code.  The types of return values should not be changed lightly.
+                    enum: [unhandled_exception, illegal_arguments, unknown_source_schema, file_already_exists, not_found]
```

Comment:
are `unknown_source_schema` and `file_already_exists` possible error codes for this operation?

Reply:
copy/paste problem, removed